### PR TITLE
8541 UI discharge confirm add submit button call to api.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#8541

*Description of changes:*
Discharge confirmation Submit button action make api call to create a discharge registration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/tests/unit/ConfirmDischarge.spec.ts [hunk 1]
```diff
@@ -116,8 +118,20 @@ describe('ConfirmDischarge registration view', () => {
...
+    // Set up for valid discharge request
+    await store.dispatch('setRegistrationNumber', '023001B')
+    await store.dispatch('setFolioOrReferenceNumber', 'A-00000402')
+    await store.dispatch('setRegistrationConfirmDebtorName', mockedDebtorNames[0])
+
     await wrapper.findComponent(DischargeConfirmSummary).vm.$emit('valid', true)
     await wrapper.findComponent(ButtonsStacked).vm.$emit('submit', true)
-    // fill in with the rest of the flow once built
+    await flushPromises()
+    expect(wrapper.vm.$route.name).toBe(RouteNames.CONFIRM_DISCHARGE)
```
Comment (kialj876) [inline ppr-ui/tests/unit/ConfirmDischarge.spec.ts]: shouldn't this have routed to the dashboard if it was successful? To ensure the mock url call isn't failing you could mock the response via sinon if you want

**Review by kialj876 — COMMENTED** [review]
Hey Doug, this looks great, but I think your test has an error

File: ppr-ui/tests/unit/ConfirmDischarge.spec.ts (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/tests/unit/ConfirmDischarge.spec.ts]: Yes, you are right, the call is failing in the test and staying at the confirm page.

Comment (doug-lovett) [reply ppr-ui/tests/unit/ConfirmDischarge.spec.ts]: Unit test updated, now routes to the dashboard.

**Comment by jordiwes** [comment]
Looks good, you have a conflict now though :)

**Review by jordiwes — APPROVED** [review]

**Pull request merged**